docs/start/install.rst

```diff
@@ -119,6 +159,104 @@ to work:
 - GCC 6+;
 - MSVC 19.14+.
 
+Build for Linux
+^^^^^^^^^^^^^^^
+- Install GCC for your distribution:
+
+  - Debian/Ubuntu:
+
+  .. code-block:: console
+
+    $ sudo apt install build-essential
+
+  - RHEL/CentOS:
+
+  .. code-block:: console
+
+    $ sudo dnf group install "Development Tools"
+
+    # or
+
+    $ sudo yum groupinstall "Development tools"
+
+
+- Then install datatable directly from github as 
+  :code:`pip install git+https://...`
+
+.. code-block:: console
+
+  $ pip install git+https://github.com/h2oai/datatable
+
+
+Build for macOS
+^^^^^^^^^^^^^^^
+
+- For macOS you will need to install the ``XCode Command Line Tools``
+- run :code:`xcode-select --install`` from your terminal and confirm the 
+  prompts for download and installation of the xcode command-line tools.
+
+.. code-block:: console
+
+  $ xcode-select --install
+
+- Then install datatable directly from github from using 
+  :code:`pip install git+https://...`
+
+.. code-block:: console
+
+  $ pip install git+https://github.com/h2oai/datatable
+
+Build for Windows
+^^^^^^^^^^^^^^^^^
+
+- In order to install the MSVC C++ compiler, you need to download and install 
+  `Visual Studio`_ or `Visual Studio Community Edition`_ and choose the option 
+  `Desktop Development with C++`, then select **install**.
+
+- Next, your shell must have the MSVC development tools available. To enable
+  them, you must run the MS Dev environment setup scripts (usually located in
+  the :code:`...\\Common7\\Tools\\` directory of your Visual Studio installation 
+  directory)
+
+  - e.g. :code:`C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\Common7\\Tools\\`
+
+- Depending on your shell, run the VS CLI environment setup script:
+
+  - Powershell
+
+  .. code-block:: powershell
+
+    > "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\Tools\Launch-VsDevShell.ps1"
+
+  - CMD or Anaconda Prompt
+
+  .. code-block:: doscon
+
+    C:\> "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\Tools\LaunchDevCmd.bat"
+
+- Lastly, On windows, datatable uses an environment variable to find the
+  MSVC compiler, so you must set that variable.
+
+  - Powershell
+
+  .. code-block:: powershell
+
+    > $env:DT_MSVC_PATH="$env:VSINSTALLDIR"+"VC\Tools\MSVC\"
+
+  - CMD or Anaconda Prompt
+
+  .. code-block:: doscon
+
+    C:\> set DT_MSVC_PATH=%VSINSTALLDIR%VC\Tools\MSVC\
```

Comment:
For VS 2019 this works out of the box: 
https://github.com/h2oai/datatable/blob/main/ci/xbuild/compiler.py#L145-L148

I guess we need to test dt with other VS versions and, if it works, add support to automatically discover VS folders other than 2019-th.